src/assert.ts

```diff
@@ -14,12 +14,12 @@ import { AssertionErrorConstructor, FailedAssertionError } from "./error.js";
  * @param expected - The expected value
  * @param errorConstructor - Constructor for the concrete error to be thrown
  */
-export function assertStringEquals(
+export function assertStringEquals<T extends string>(
```

Comment:
What is the objective of this generic that extends string? If the aim is to enforce strong typing, we could do that through type assertion check inside the function right?

Reply:
This is pure TypeScript developer experience.

By adding the `asserts actual is T` as function return type, we make the assertions of this function available to TypeScript's control flow analysis, such that calling code will then understand that the type of `expected` is a `string`.

This is explained by TypeScript's lead architect here: https://github.com/microsoft/TypeScript/pull/32695

I added this in because it's simple to add, and makes using these functions a bit nicer in TypeScript.